Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1632,38 +1808,40 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes(GridLayoutState& gridL
 
 void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizesMasonry(GridLayoutState& gridLayoutState)
 {
-    if (m_strategy->isComputingSizeContainment()) {
+    if (m_strategy->isComputingSizeContainment() || !m_grid.hasGridItems()) {
         handleInfinityGrowthLimit();
         return;
     }
 
-    Vector<GridTrack>& allTracks = tracks(m_direction);
-    Vector<GridItemWithSpan> itemsSortedByIncreasingSpan;
-    Vector<GridItemWithSpan> itemsCrossingFlexibleTracks;
-    SingleThreadWeakHashSet<RenderBox> itemsSet;
-    MasonryIndefiniteItems masonryIndefiniteItems;
+    StdMap<unsigned, MasonryMinMaxTrackSize> indefiniteTrackSizes;
```

Comment:
The value here seems to come from the intrinsic sizes of grid items for a particular span length and I don't think `indefiniteTrackSizes` is that

Reply:
Updated the name